Setup for doc-builder and docs for TGI (#740)
huggingface/text-generation-inference: docs/source/supported_models.md

> # Supported Models and Hardware

Reviewer: It'd be nice to have a brief sentence or two here about this section instead of leaving it empty. Maybe something like:

"TGI enables serving optimized models on specific hardware for the highest performance. The following sections list which models are hardware are supported."

## Supported Models

> List of optimized models are below.

Reviewer: ```suggestion
The following models are optimized and can be served with TGI:
```

- [BLOOM](https://huggingface.co/bigscience/bloom)
- [FLAN-T5](https://huggingface.co/google/flan-t5-xxl)
- [Galactica](https://huggingface.co/facebook/galactica-120b)
- [GPT-Neox](https://huggingface.co/EleutherAI/gpt-neox-20b)
- [Llama](https://github.com/facebookresearch/llama)
- [OPT](https://huggingface.co/facebook/opt-66b)
- [SantaCoder](https://huggingface.co/bigcode/santacoder)
- [Starcoder](https://huggingface.co/bigcode/starcoder)
- [Falcon 7B](https://huggingface.co/tiiuae/falcon-7b)
- [Falcon 40B](https://huggingface.co/tiiuae/falcon-40b)
- [MPT](https://huggingface.co/mosaicml/mpt-30b)
- [Llama V2](https://huggingface.co/meta-llama)

> If the above list lacks the model you would like to serve, depending on the model's pipeline type, you can try to initialize and serve the model on best-effort basis like below:

Reviewer: Hmm what exactly does "best-effort" basis mean?

Author: It means that you should try and see if it works well, sort of. i.e. we don't promise the user anything for models that aren't optimized.

> `AutoModelForCausalLM.from_pretrained(<model>, device_map="auto")`

Reviewer: Maybe throw these in a code block?

Author: You mean, the list, or the small AutoModel instantiation snippets?

Author: Done.

or

`AutoModelForSeq2SeqLM.from_pretrained(<model>, device_map="auto")`. 

> For the optimized models above, TGI uses custom CUDA kernels for better inference. You can add the flag `--disable-custom-kernels` at the end of the `docker run` command if you wish to disable them.

Reviewer: I think it'd be nicer to move this info to the beginning of the section so users know everything upfront.

Author: Can you elaborate? I think given this library is optimizing above architectures specifically, i.e. it's the library's trademark, it's better if we provide the optimized models first.

Author: Good idea!

## Supported Hardware

> Text Generation Inference optimized models supported on NVIDIA [A100](https://www.nvidia.com/en-us/data-center/a100/), [A10G](https://www.nvidia.com/en-us/data-center/products/a10-gpu/) and [T4](https://www.nvidia.com/en-us/data-center/tesla-t4/) GPUs with CUDA 11.8+. Note that you have to install [NVIDIA Container Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html) to use it. For other hardware, continuous batching will still apply, but you might observe downgrades on some of the operations (e.g. flash attention, paged attention) will not be executed. 

Reviewer: ```suggestion
TGI optimized models are supported on NVIDIA [A100](https://www.nvidia.com/en-us/data-center/a100/), [A10G](https://www.nvidia.com/en-us/data-center/products/a10-gpu/) and [T4](https://www.nvidia.com/en-us/data-center/tesla-t4/) GPUs with CUDA 11.8+. Note that you have to install [NVIDIA Container Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html) to use it. For other hardware, continuous batching will still apply, but some operations like flash attention and paged attention will not be executed. 
```

TGI is also supported on the following AI hardware accelerators:

> - *Habana first-gen Gaudi and Gaudi2:* checkout [here](https://github.com/huggingface/optimum-habana/tree/main/text-generation-inference) how to serve models with TGI on Gaudi and Gaudi2 with [Optimum Habana](https://huggingface.co/docs/optimum/habana/index)

Reviewer: ```suggestion
- *Habana first-gen Gaudi and Gaudi2:* check out this [example](https://github.com/huggingface/optimum-habana/tree/main/text-generation-inference) how to serve models with TGI on Gaudi and Gaudi2 with [Optimum Habana](https://huggingface.co/docs/optimum/habana/index)
```
